src/Functions/Public/Connect-vRAServer.ps1

```diff
@@ -176,24 +162,31 @@
                     }
 
                     # --- Logging in with a domain
-                    if (@("SAM", "SAMAccountName").Contains($UserAttribute) -and $Username -match '@') {
+                    $URI = "https://$($Server)/csp/gateway/am/idp/auth/login?access_token"
+                    if ($Username -match '@') {
                         # Log in using the advanced authentication API
-                        $URI = "https://$($Server)/csp/gateway/am/idp/auth/login?access_token"
                         $User = $Username.split('@')[0]
                         $Domain = $Username.split('@')[1]
                         $JSON = @{
```

Comment:
Given that `JSON` and `AlternativeJson` hastables have the same structure, was wondering if we could do something clever and reuse a single hashtable but overwrite properties `if` they need updating.

Reply:
updated the code, take a look, should show $RawBody in place of $JSON variables, then I convert on the call to Invoke